module/entityAssessors/helper.js

```diff
@@ -193,17 +193,30 @@ module.exports = class entityAssessorHelper {
                     name: { $in: Object.values(entityTypeQueryList) }
                 }, { name: 1 }).lean();
 
-                let userIds = assessorData.map(assessor => assessor.userId);
+                let userExternalIds = assessorData.map(assessor => assessor.externalId);
```

Comment:
Why loop through the same assessorData again and again, why not do it all in one loop ?

Line 196, 197, line 162, line 229.

@ajithtunerlabs

Reply:
addressed.